rpc/resource/resourceext/resource.go

```diff
@@ -0,0 +1,103 @@
+package resourceext
```

Comment:
could this just live in the `resource` package? What are the benefits of putting it inside another package?

Reply:
This is common pattern to separate interface from implementations. See [package image](https://godoc.org/image) or [hash package](https://godoc.org/hash)  for example.